src/main/java/com/suse/saltstack/netapi/event/EventStream.java

```diff
@@ -3,113 +3,109 @@
+            initializeStream(config);
+        } catch (SaltStackException e) {
+            e.printStackTrace();
+        }
     }
 
     /**
-     * An alternate constructor used to create this object.  Does not
-     * automatically start event processing.  This constructor is used for
-     * unit testing only and allows for the injection of an existing and
-     * possibly pre-configured {@link Client} object.
-     * @param config Contains the necessary details such as endpoint URL and
-     *               authentication token required to create the request to obtain
-     *               the {@link EventInput} event stream.
-     * @param client Passed client to use when configuring the event stream
+     * This method initiates the WebSocket handshake and its life.
+     * This method is intended for use during unit testing only!
+     * End users should not call this methods directly.
+     *
+     * @param uri WebSocket URI
+     * @throws DeploymentException If annotatedEndpoint instance is not valid.
+     * @throws IOException If WebSocket connection to remote server fails.
      */
-    public EventStream(ClientConfig config, Client client) {
-        client.register(new SseFeature());
-        WebTarget target = client.target(config.get(ClientConfig.URL) + "/events");
-        Invocation.Builder builder = target.request(new MediaType("text",
-                "event-stream"));
-        builder.header("X-Auth-Token", config.get(ClientConfig.TOKEN));
-
-        eventInput = builder.get(EventInput.class);
+    public void processEvents(URI uri) throws DeploymentException, IOException {
+        synchronized (websocketContainer) {
+            this.session = websocketContainer.connectToServer(this, uri);
+        }
     }
 
     /**
-     * This method initiates the event processing loop. This method is intended for
-     * use during unit testing only!  End users should not call this methods directly.
-     */
-    public void processEvents() {
-        if (!eventProcessingStarted) {
-            if (eventInput != null && !eventInput.isClosed()) {
-                eventProcessingStarted = true;
-                while (!eventInput.isClosed()) {
-                    final InboundEvent inboundEvent = eventInput.read();
-                    if (inboundEvent == null) {
-                        // connection has been closed
-                        close();
-                        break;
-                    }
-                    synchronized (listeners) {
-                        for (EventListener listener : listeners) {
-                            listener.notify(inboundEvent.readData(String.class));
-                        }
-                    }
-                }
+     * Connect the WebSocket to the server pointing to /ws/{token} to receive events.
+     */
+    private void initializeStream (ClientConfig config) throws SaltStackException {
+        try {
+            URI uri = new URI(config.get(ClientConfig.URL).toString()
```

Comment:
I think something like this should be more robust than string replacement.
```java
new URI("ws", uri.getSchemeSpecificPart(), uri.getFragment())
    .resolve("/ws/" + config.get(ClientConfig.TOKEN)))
```
In the long term i think it makes sense to have only hostname and port in the config instead of a whole URI. But that is out of scope for this PR. :smile: